crates/fullnode/src/runner.rs

```diff
@@ -300,6 +301,9 @@ where
             }
         };
 
+        let (proven_commitments, _) =
+            filter_out_proven_commitments(&self.ledger_db, &proven_commitments)?;
```

Comment:
and this should be filtered by index from zk output `preprover_commitments` of sorted list of sequencer commitments on the da slot